src/main/java/com/faforever/client/update/CheckForUpdateTask.java

```diff
@@ -2,101 +2,108 @@
 
 import com.faforever.client.config.ClientProperties;
 import com.faforever.client.i18n.I18n;
+import com.faforever.client.preferences.PreferencesService;
 import com.faforever.client.task.CompletableTask;
-import com.google.common.io.CharStreams;
-import com.google.common.reflect.TypeToken;
-import com.google.gson.FieldNamingPolicy;
+import com.faforever.client.update.ClientConfiguration.ReleaseInfo;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
+import lombok.SneakyThrows;
 import org.apache.maven.artifact.versioning.ComparableVersion;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.config.ConfigurableBeanFactory;
 import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Component;
 
-import javax.inject.Inject;
-import java.io.InputStreamReader;
-import java.io.Reader;
+import java.io.IOException;
 import java.lang.invoke.MethodHandles;
-import java.lang.reflect.Type;
-import java.net.HttpURLConnection;
 import java.net.URL;
-import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.List;
+import java.util.regex.Pattern;
 
 @Component
 @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 public class CheckForUpdateTask extends CompletableTask<UpdateInfo> {
 
   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-  private final Gson gson;
+  private static final Pattern SEMVER_PATTERN = Pattern.compile("v\\d+(\\.\\d+)*[^.]*");
 
-  //TODO: switch to constructor injection, superclass CompletableTask<UpdateInfo> has no default constructor
-  @Inject
-  ClientProperties clientProperties;
-  @Inject
-  I18n i18n;
+  private final I18n i18n;
+  private final PreferencesService preferencesService;
 
   private ComparableVersion currentVersion;
 
-  public CheckForUpdateTask() {
+  @VisibleForTesting
+  FileSizeReader fileSizeReader = url -> url
```

Comment:
Shouldn't this be injected as a bean? Then @VisibleForTesting would be obsolete as well.

Reply:
hmm well, that would work but it's not a component that should generally be available as a singleton but a very local solution with the sole purpose that I can mock the mechanism in the unit test.

However, there is in fact other code that could use it as well (`com.faforever.client.map.MapService#getFileSize`) which supports your point. I'll add a TODO and refactor this at another time